plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/editor/context/project/EncoderServer.kt

```diff
@@ -118,10 +118,12 @@ class EncoderServer(val project: Project) : Disposable {
         val map = mutableMapOf<String, String>(
             "PORT" to port.toString(),
             "START_AMAZONQ_LSP" to "true",
-            "Q_WORKER_THREADS" to threadCount.toString(),
             "CACHE_DIR" to cachePath.toString(),
             "MODEL_DIR" to cachePath.resolve("qserver").toString()
         )
+        if (threadCount > 0 && threadCount < 100) {
+            map["Q_WORKER_THREADS"] = threadCount.toString()
+        }
         if (isGpuEnabled) {
```

Comment:
unrelated, but why doesn't `Q_ENABLE_GPU = false` work?

Reply:
The lsp does not handle strings well, we are going to improve support on that side but at the moment this is the best fix we think.